libraries/AP_Scripting/examples/copter-fast-descent.lua

```diff
@@ -0,0 +1,158 @@
+-- Copter descends very rapidly in a spiral pattern to a present altitude above home
+--
+-- CAUTION: This script only works for Copter
+-- this script waits for the vehicle to be changed to Guided mode and then:
+--    a) flies a spiral pattern using the velocity and acceleration control
+--    b) slows the spiral and stops at the preset altitude
+--    c) switches to RTL
+
+-- constants
+local copter_guided_mode_num = 4    -- Guided mode is 4 on copter
+local copter_rtl_mode_num = 6       -- RTL is 6 on copter
+local alt_above_home_min = 50       -- copter will stop at this altitude above home
+local circle_radius_rate_max_ms = 1 -- radius expands at max of this many m/s
+local circle_radius_accel_mss = 1   -- radius expansion speed accelerates at this many m/s/s
+local circle_radius_max = 30        -- target circle's maximum radius
+local speed_xy_max = 5              -- max target horizontal speed
+local accel_xy = 1                  -- horizontal acceleration in m/s^2
+local speed_z_max = 10              -- target descent rate is 15m/s
+local accel_z = 1                   -- target vertical acceleration is 1m/s/s
+
+-- timing and state machine variables
+local stage = 0                     -- stage of descent
+local last_update_ms = 0            -- system time of last update
+local dt = 0.01                     -- update rate of script (0.01 = 100hz)
+local interval_ms = 1               -- update interval in ms
+local last_print_ms = 0             -- pilot update timer
+
+-- control related variables
+local circle_center_pos = Vector3f()-- center of circle position as an offset from EKF origin
+local circle_radius = 0             -- target circle's current radius (this is slowly expanded to circle_radius_max)
+local circle_radius_rate_ms = 0     -- target circle's radius is increasing at this rate in m/s
+local circle_angle_rad = 0          -- current target angle on circle (in radians)
+local target_alt_D = 0              -- target altitude in m from EKF origin (Note: down is positive)
+local speed_xy = 0                  -- target horizontal speed (i.e. tangential velocity or horizontal speed around the circle)
+local speed_z = 0                   -- target descent rate currently
+local target_yaw_deg = 0            -- target yaw in degrees (degrees is more convenient based on interface)
+
+-- the main update function
+function update()
+
+  -- update dt
+  local now_ms = millis()
+  if (last_update_ms ~= 0) then
```

Comment:
if you don't init `last_update_ms` to 0 you can just check for nill rather than checking for 0. So the if would just be:
```
  if last_update_ms then
```
not that it makes any difference.

Reply:
ok, removed init of last-update-ms. txs